web/elm/src/SideBar/SideBar.elm

```diff
@@ -96,7 +96,16 @@ update message model =
         Click SideBarResizeHandle ->
             ( { model | draggingSideBar = True }, [] )
 
-        Click (SideBarStarIcon pipelineID) ->
+        Click (SideBarFavoritedIcon pipelineID) ->
+            let
+                favoritedPipelines =
+                    toggle pipelineID model.favoritedPipelines
+            in
+            ( { model | favoritedPipelines = favoritedPipelines }
+            , [ Effects.SaveFavoritedPipelines <| favoritedPipelines ]
+            )
+
+        Click (PipelineCardFavoritedIcon pipelineID) ->
```

Comment:
Also we need to refactor this part. It's kinda confusing to see the `PipelineCardFavoritedIcon` here.

Reply:
you're doing my job for me, i love it